datasets/xfun/xfun.py

```diff
@@ -0,0 +1,298 @@
+import json
+import logging
+import os
+
+import numpy as np
+from PIL import Image
+from transformers import AutoTokenizer
+
+import datasets
+
+
+_CITATION = """
+@article{Xu2020LayoutXLMMP,
+  title         = {LayoutXLM: Multimodal Pre-training for Multilingual Visually-rich Document Understanding},
+  author        = {Yiheng Xu and Tengchao Lv and Lei Cui and Guoxin Wang and Yijuan Lu and Dinei Florencio and Cha Zhang and Furu Wei},
+  year          = {2021},
+  eprint        = {2104.08836},
+  archivePrefix = {arXiv},
+  primaryClass  = {cs.CL}
+}
+"""
+
+_DESCRIPTION = """
+XFUND is a multilingual form understanding benchmark dataset that includes human-labeled forms with key-value pairs in 7 languages (Chinese, Japanese, Spanish, French, Italian, German, Portuguese).
+"""
+
+_VERSION = datasets.Version("1.0.0")
+
+_URL = "https://github.com/doc-analysis/XFUN/releases/download/v1.0/"
+_LANG = ["zh", "de", "es", "fr", "it", "ja", "pt"]
+
+logger = logging.getLogger(__name__)
+
+
+def normalize_bbox(bbox, size):
+    return [
+        int(1000 * bbox[0] / size[0]),
+        int(1000 * bbox[1] / size[1]),
+        int(1000 * bbox[2] / size[0]),
+        int(1000 * bbox[3] / size[1]),
+    ]
+
+
+def simplify_bbox(bbox):
+    return [
+        min(bbox[0::2]),
+        min(bbox[1::2]),
+        max(bbox[2::2]),
+        max(bbox[3::2]),
+    ]
+
+
+def merge_bbox(bbox_list):
+    x0, y0, x1, y1 = list(zip(*bbox_list))
+    return [min(x0), min(y0), max(x1), max(y1)]
+
+
+def load_image(image_path):
+    with Image.open(image_path) as image:
+        w, h = image.size
+        image = image.resize((224, 224), resample=Image.BILINEAR)
+        image_arr = np.asarray(image.copy())  # copy to make it writeable
+        image_arr = image_arr[:, :, ::-1]  # BGR -> RGB
+        image_arr = np.transpose(image_arr, axes=(2, 0, 1))
+    return image_arr, (w, h)
```

Comment:
It's better to return a `PIL.Image` object here, change the `"image"` feature type from `datasets.Array3D(shape=(3, 224, 224), dtype="uint8")` to `datasets.Image()` and add a Dataset Usage section with a map function (and mention this transform is needed to reproduce the results from the paper):
```python
def resize_and_convert_to_array(batch)
    image_arrs = []
    for img in batch["image"]:        
        image = image.resize((224, 224), resample=Image.BILINEAR)
        image_arr = np.asarray(image.copy())  # copy to make it writeable
        image_arr = image_arr[:, :, ::-1]  # BGR -> RGB
        image_arr = np.transpose(image_arr, axes=(2, 0, 1))
        image_arrs.append(image_arr)
    batch["image"] = image_arrs
    return batch

ds_features = ds.features.copy()
ds_features["image"] = datasets.Array3D(shape=(3, 224, 224), dtype="uint8")
ds = ds.map(resize_and_convert_to_array, batch_size=100, batched=True, writer_batch_size=100, num_proc=4, features=ds_features)
```